src/overlays/actors/ovl_En_Onpuman/z_en_onpuman.c

```diff
@@ -27,29 +28,150 @@ const ActorInit En_Onpuman_InitVars = {
+    Collider_InitAndSetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actor.targetMode = 6;
+    this->unk_2A4 = 0;
+    this->unk_2A0 = NULL;
+    this->actionFunc = func_80B121D8;
+    this->actor.velocity.y = -10.0f;
+    this->actor.terminalVelocity = -10.0f;
+    this->actor.gravity = -5.0f;
+}
+
+void EnOnpuman_Destroy(Actor* thisx, PlayState* play) {
+    EnOnpuman* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+Actor* func_80B11F44(PlayState* play) {
+    Actor* itemAction = play->actorCtx.actorLists[ACTORCAT_ITEMACTION].first;
+
+    if (itemAction != NULL) {
+        if (itemAction->id != ACTOR_EN_GAKUFU) {
+            itemAction = itemAction->next;
+        }
+        return itemAction;
+    }
+    return NULL;
+}
+
+void func_80B11F78(EnOnpuman* this, PlayState* play) {
+    if (play->msgCtx.ocarinaMode == 4) {
+        this->actionFunc = func_80B121D8;
+        if (this->actor.cutscene != -1) {
+            ActorCutscene_Stop(this->actor.cutscene);
+        }
+    } else if (play->msgCtx.ocarinaMode == 3) {
+        play_sound(NA_SE_SY_CORRECT_CHIME);
+        play->msgCtx.ocarinaMode = 4;
+        if (this->actor.cutscene != -1) {
+            ActorCutscene_Stop(this->actor.cutscene);
+        }
+        this->actionFunc = func_80B121D8;
+    }
+}
+
+void func_80B1202C(EnOnpuman* this, PlayState* play2) {
+    PlayState* play = play2;
+
+    if ((Message_GetState(&play->msgCtx) == 5) && (Message_ShouldAdvance(play))) {
+        switch (play->msgCtx.currentTextId) {
+            case 0x8D4:
+                this->unk_2A4 |= 1;
+                func_80151938(play, 0x8DA);
+                break;
+
+            case 0x8DA:
+                func_80151938(play, 0x8D6);
+                if (this->unk_2A0 != NULL) {
+                    this->unk_2A0->home.rot.x = 0x50;
+                }
+                break;
+
+            case 0x8D6:
+                this->actionFunc = func_80B11F78;
+                func_80152434(play, 0x3A);
+                if (this->unk_2A0 != NULL) {
+                    this->unk_2A0->home.rot.x = 0;
+                }
+                break;
+        }
+    }
+    if (this->unk_2A4 & 1) {
+        if (this->actor.cutscene == -1) {
+            this->unk_2A4 &= ~1;
+        } else if (ActorCutscene_GetCanPlayNext(this->actor.cutscene)) {
+            this->unk_2A4 &= ~1;
+            ActorCutscene_StartAndSetUnkLinkFields(this->actor.cutscene, &this->actor);
+        } else {
+            ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+        }
+    }
+}
+
+void func_80B1217C(EnOnpuman* this, PlayState* play) {
+    if ((Message_GetState(&play->msgCtx) == 5) && (Message_ShouldAdvance(play))) {
+        this->actionFunc = func_80B121D8;
+        func_801477B4(play);
+    }
+}
+
+void func_80B121D8(EnOnpuman* this, PlayState* play) {
+    s16 absoluteValue;
```

Comment:
This does get assigned to the absolute value, and it would probably be better to name it something more specific. In this case we see its the diff between yaw, and shape.rot.y, which normally we just call `yaw`.